packages/core/src/modules/dids/__tests__/IndyDidResolver.test.ts

```diff
@@ -0,0 +1,111 @@
+import type { GetNymResponse } from 'indy-sdk'
+
+import { mockFunction } from '../../../../tests/helpers'
+import { JsonTransformer } from '../../../utils/JsonTransformer'
+import { IndyLedgerService } from '../../ledger/services/IndyLedgerService'
+import { parseDid } from '../parse'
+import { IndyDidResolver } from '../resolvers/IndyDidResolver'
+
+import didSovR1xKJw17sUoXhejEpugMYJFixture from './__fixtures__/didSovR1xKJw17sUoXhejEpugMYJ.json'
+import didSovWJz9mHyW9BZksioQnRsrAoFixture from './__fixtures__/didSovWJz9mHyW9BZksioQnRsrAo.json'
+
+jest.mock('../../ledger/services/IndyLedgerService')
+const IndyLedgerServiceMock = IndyLedgerService as jest.Mock<IndyLedgerService>
+
+const getParsed = (did: string) => {
+  const parsed = parseDid(did)
+
+  if (!parsed) {
+    throw new Error('Could not parse')
+  }
+
+  return parsed
+}
+
+describe('DidResolver', () => {
+  describe('IndyDidResolver', () => {
+    let ledgerService: IndyLedgerService
+    let indyDidResolver: IndyDidResolver
+
+    beforeEach(() => {
+      ledgerService = new IndyLedgerServiceMock()
+      indyDidResolver = new IndyDidResolver(ledgerService)
+    })
+
+    it('should correctly resolve a did:sov document', async () => {
+      const did = 'did:sov:R1xKJw17sUoXhejEpugMYJ'
+
+      const nymResponse: GetNymResponse = {
+        did: 'R1xKJw17sUoXhejEpugMYJ',
+        verkey: 'E6D1m3eERqCueX4ZgMCY14B4NceAr6XP2HyVqt55gDhu',
+        role: 'ENDORSER',
+      }
+
+      const endpoints: IndyEndpointAttrib = {
+        endpoint: 'https://ssi.com',
+        profile: 'https://profile.com',
+        hub: 'https://hub.com',
+      }
+
+      mockFunction(ledgerService.getPublicDid).mockResolvedValue(nymResponse)
+      mockFunction(ledgerService.getEndpointsForDid).mockResolvedValue(endpoints)
+
+      const result = await indyDidResolver.resolve(did, getParsed(did))
+
+      expect(JsonTransformer.toJSON(result)).toMatchObject({
+        didDocument: didSovR1xKJw17sUoXhejEpugMYJFixture,
+        didDocumentMetadata: {},
+        didResolutionMetadata: {
+          contentType: 'application/did+ld+json',
+        },
+      })
+    })
+
+    it('should resolve a did:sov document with routingKeys and types entries in the attrib', async () => {
+      const did = 'did:sov:WJz9mHyW9BZksioQnRsrAo'
+
+      const nymResponse: GetNymResponse = {
+        did: 'WJz9mHyW9BZksioQnRsrAo',
+        verkey: 'GyYtYWU1vjwd5PFJM4VSX5aUiSV3TyZMuLBJBTQvfdF8',
+        role: 'ENDORSER',
+      }
+
+      const endpoints: IndyEndpointAttrib = {
+        endpoint: 'https://agent.com',
+        types: ['endpoint', 'did-communication', 'DIDComm'],
+        routingKeys: ['routingKey1', 'routingKey2'],
+      }
+
+      mockFunction(ledgerService.getPublicDid).mockReturnValue(Promise.resolve(nymResponse))
+      mockFunction(ledgerService.getEndpointsForDid).mockReturnValue(Promise.resolve(endpoints))
+
+      const result = await indyDidResolver.resolve(did, getParsed(did))
```

Comment:
We need to parse it on a higher level because we need to extract the did method from the did. So we could parse it in the did resolver service, not pass it and then parse the did again in the indy did resolver.